plugins/Files/js/sagas/helpers.js

```diff
@@ -138,22 +138,12 @@ const bytesPerGB = new BigNumber('1000000000')
 // `duration` is in blocks, size is in GB.
 // returns a `BigNumber` representing the average number of Siacoins per GB per duration
 export const estimatedStoragePriceGBSC = (hosts, size, duration) => {
-	const nhosts = 36
-	let storagePrices = List(hosts).map((host) => new BigNumber(host.storageprice))
-
-	// Take the 36 cheapest hosts from the list
-	storagePrices = storagePrices.sort((p1, p2) => {
-		if (p2.greaterThan(p1)) {
-			return -1
-		}
-		if (p2.lessThan(p1)) {
-			return 1
-		}
-		return 0
-	}).take(nhosts)
-
-	// Sum and average the storage prices
-	const averagePricePerByteBlockH = storagePrices.reduce((sum, price) => sum.add(price), new BigNumber(0)).dividedBy(4)
+	const minimumHosts = 14
+	const hostPrices = List(hosts).map((host) => new BigNumber(host.storageprice))
+	if (hostPrices.size < minimumHosts) {
+		throw 'not enough hosts'
+	}
+	const averagePricePerByteBlockH = hostPrices.reduce((sum, price) => sum.add(price), new BigNumber(0)).dividedBy(hostPrices.size).times(9)
```

Comment:
and probably a TODO explaining that this functionality should really be handled by the API, but no endpoint yet exists.